src/main/java/bio/terra/app/controller/AdminApiController.java

```diff
@@ -124,4 +127,29 @@ public ResponseEntity<SnapshotModel> adminRetrieveSnapshot(UUID id) {
     SnapshotModel snapshotModel = snapshotService.retrieveSnapshotModel(id, include, userReq);
     return ResponseEntity.ok(snapshotModel);
   }
+
+  @Override
+  public ResponseEntity<JobModel> adminInheritSteward(UUID id, Boolean inheritSteward) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    logger.info(
+        "Verifying resource type admin authorization: {} for resource type: {} and resource id: {}",
+        userReq.getEmail(),
+        IamResourceType.DATASET,
+        id);
+    iamService.verifyResourceTypeAdminAuthorized(
+        userReq, IamResourceType.DATASET, IamAction.ADMIN_TOGGLE_INHERIT_STEWARD);
+
+    // dataset already has the requested value for inheritSteward
+    if (datasetService.retrieveDatasetSummary(id).isInheritSteward() == inheritSteward) {
+      throw new BadRequestException(
+          String.format("Dataset %s already has inheritSteward set to %s", id, inheritSteward));
```

Comment:
Since `PUT` is idempotent returning `400` when the value is set to the same value is not correct, you should return `200` here.

Reply:
Since the response type is a job model, how would I do this? Like ResponseEntity.ok(null) ?

Reply:
Oh, I forgot that the result is a job. I think `204` is the correct response for a `PUT` with no effect.
```suggestion
      return ResponseEntity.noContent().build();
```

And you'd have to update the openapi yaml to have a 204 response section.